scipy/sparse/_data.py

```diff
@@ -273,6 +278,11 @@ def _arg_min_or_max(self, axis, out, op, compare):
                 am = op(mat.data)
                 m = mat.data[am]
 
+                if explicit:
+                    if m == 0:
+                        return []
```

Comment:
This seems wrong to me. We'll only get `m == 0` in the case where an explicit zero was stored, and it seems to me that an explicit zero is a valid index for an argmin/argmax operation to return, even with `explicit=True`.

Reply:
Thanks @perimosocordiae for the review.  I have a question regarding the behaviour we want for argmin/max when explicit is True: at the moment, we have
        assert_equal(mat.argmin(explicit=True), np.argmin(mat.data))
        assert_equal(mat.argmax(explicit=True), np.argmax(mat.data))
If that's correct, we should raise a ValueError when axis is Noneand explicit is True for zero matrices because mat.data would be an empty array in that case. I have changed the code to do that but let me know if I misunderstood

Reply:
I have added a simple benchmark for argmax just to make sure my changes does not have adverse impact:
Before changes:
[100.00%] ··· sparse.Argmax.time_argmax                                                   ok
[100.00%] ··· ========= ============= ============= =============
              --                          format                 
              --------- -----------------------------------------
               density       csr           csc           coo     
              ========= ============= ============= =============
                 0.01      699±50μs    1.10±0.02ms    14.0±0.2μs 
                 0.1     4.88±0.07ms   7.77±0.03ms     137±4μs   
                 0.5      24.0±0.5ms    34.2±0.9ms   2.37±0.05ms 
              ========= ============= ============= =============
After changes:
[100.00%] ··· sparse.Argmax.time_argmax                                                   ok
[100.00%] ··· ========= ============= ============= ============
              --                         format                 
              --------- ----------------------------------------
               density       csr           csc          coo     
              ========= ============= ============= ============
                 0.01      100±2μs      97.0±0.5μs   12.6±0.4μs 
                 0.1       439±3μs       437±3μs      64.9±3μs  
                 0.5     2.43±0.05ms   2.43±0.02ms    315±10μs  
              ========= ============= ============= ============
I think the time improvement are simply coming from the removal of sum_duplicates() which looks expensive

Reply:
@perimosocordiae Regarding the test case: I mentioned above
assert_equal(mat.argmin(explicit=True), np.argmin(mat.data))
I now realise I forgot to take into account the non canonical case where it makes sense to return 'am'. I have added a simple test with a 2 by 2 zero matrix to make sure this works.